Assets/Scripts/R&D Tree/ResearchNode.cs

```diff
@@ -0,0 +1,102 @@
+using UnityEngine;
+using UnityEngine.Events;
+
+public class ResearchNode : MonoBehaviour
+{
+    [SerializeField] private RDTreeManager _treeManager;
```

Comment:
Ultimately speaking, you would create a ScriptableObject that holds the information about the node. The information should be put in a generic node class, holding functions and callbacks, which can be set when the tree is initialized. Your node should just call the callback function when it needs to, but shouldn't know the implementation of the callback or from whom it came from. 

Atm, you're creating a circle dependency with the treemanager calling functions in the node and the node calling functions in the treemanager... Rather create a serialized list in the treemanager and add them there by hand. Create an Action in the node which can be used as a callback when attaching the node to the tree. This will be called when the research is done.

You can leave it for now, especially cause the deadline is creeping up, but try to think and/or draw out the structure and their connections. Visualizing it might help you write even better code! Dependencies and the structure will be even more important with the layer system you're currently working on.